pkgs/top-level/all-packages.nix

```diff
@@ -38866,9 +38866,12 @@ with pkgs;
 
   pymol = callPackage ../applications/science/chemistry/pymol { };
 
-  quantum-espresso = callPackage ../applications/science/chemistry/quantum-espresso { };
-
-  quantum-espresso-mpi = callPackage ../applications/science/chemistry/quantum-espresso { useMpi = true; };
+  quantum-espresso = callPackage ../applications/science/chemistry/quantum-espresso {
+    hdf5 = hdf5.override {
```

Comment:
This is already defined ad `hdf5-fortran`. You could just use `hdf5-fortran` directly as an input for quantum-espresso